small_lms_train.py

```diff
@@ -14,9 +16,11 @@
 # -----------------------------
 
 # setup cuda
-torch.cuda.memory._record_memory_history()
-device = "cuda" if torch.cuda.is_available() else "cpu"
-torch.set_default_device("cpu")
+if torch.cuda.is_available():
+  torch.cuda.memory._record_memory_history()
+  device = "cuda" 
+else: device="cpu"
+torch.set_default_device(device)
```

Comment:
Tensors in the dataloaders need to be on the `cpu` for the training loop to run. Even when using `cuda`. If default devise is `cuda` then tokenizer will move all created tensors there. Can just remove this line